app/scripts/SinkhornNNLSLinseedC.R

```diff
@@ -607,13 +607,25 @@ SinkhornNNLSLinseed <- R6Class(
       Ae <- restored[indice, ]
       ## X
       self$init_X <- Ae %*% t(self$R)
-      ## D
-      self$init_D_h <- ginv(t(self$init_X)) %*% self$A
-      self$init_D_w <- self$init_D_h * (self$M/self$N)
-      ## Omega
-      V__ <- self$S %*% self$V_row %*% t(self$R)
-      self$init_Omega <- V__ %*% ginv(diag(self$init_D_w[,1]) %*% self$init_X)
+      self$setInitParamsByX()
+    },
+
+    selectInitXFixed = function(gene_list) {
+      ## X
+      self$init_X <- self$new_points[gene_list, ]
+      self$setInitParamsByX()
+    },
+
+    selectInitXMarkerMeans = function(markers_list) {
+      marker_means <- lapply(markers_list, function(x) {
+        sel <- x[x %in% rownames(self$new_points)]
+        if (sum(sel) == 0) stop("One or more cell types has zero markers present")
+        colMeans(self$new_points[sel, ])
```

Comment:
Essentially, markers initialisation is here. For the set of marker genes for each cell type, we take the mean of the corresponding self$new_points column, and that is the X. Judging by the plots, it works fine, and proportions, generated from that seem to be good.